src/protocols/protocol_block_sync.cpp

```diff
@@ -60,7 +61,7 @@ void protocol_block_sync::start()
     protocol_timer::start(monitor_interval, BIND1(handle_event, _1));
 
     chain_.subscribe_headers(BIND4(handle_reindexed, _1, _2, _3, _4));
-    SUBSCRIBE2(block, handle_receive_block, _1, _2);
+    SUBSCRIBE3(block, handle_receive_block, bitcoin_settings_, _1, _2);
```

Comment:
It seems that protocols should be constructed with settings, not passed on each handler invoke.

Reply:
Agreed. Fixed.